src/pages/Auth/Auth.vue

```diff
@@ -0,0 +1,114 @@
+<script>
+export default {
+  data() {
+    return {
+      previousHeight: 0
+    }
+  },
+  methods: {
+    afterEnter(element) {
+      element.style.height = 'auto'
+    },
+    beforeLeave(element) {
+      this.previousHeight = getComputedStyle(element).height
+    },
+    enter(element) {
+      const { height } = getComputedStyle(element)
+      element.style.height = this.previousHeight
+
+      setTimeout(() => {
+        element.style.height = height
+      })
+    }
```

Comment:
Just a question, what would be the difference if you didn't put it in the `setTimeout`? As this doesn't have a delay, I would imagine it would run immediately?

Reply:
Yeah there are a few reasons we might want to do this (and to be honest I stole this from animations I wrote for the universal login application about a year ago so I can't remember which reason this was for, likely all of them):
1) Using a null timeout allows us to put this piece of logic in the event queue but _not_ execute it immediately
2) It enables next-tick-like css animations here, being that line 17 is caught by the renderer, executed, and then line 20 is rendered after. Without the timeout, line 17 is ignored by the renderer.